pkg/resolution/resolver/git/resolver.go

```diff
@@ -247,9 +251,15 @@ var _ framework.TimedResolution = &Resolver{}
 // GetResolutionTimeout returns a time.Duration for the amount of time a
 // single git fetch may take. This can be configured with the
 // fetch-timeout field in the git-resolver-config configmap.
-func (r *Resolver) GetResolutionTimeout(ctx context.Context, defaultTimeout time.Duration) time.Duration {
-	conf := framework.GetResolverConfigFromContext(ctx)
-	if timeoutString, ok := conf[DefaultTimeoutKey]; ok {
+func (r *Resolver) GetResolutionTimeout(ctx context.Context, defaultTimeout time.Duration, params map[string]string) time.Duration {
+	gitResolverConfig := GetGitConfig(ctx)
+	var conf ScmInfo
+	if tokenIdentifier, ok := params[TokenIdentifierParam]; ok {
+		conf = gitResolverConfig.ScmTokens[tokenIdentifier]
+	} else {
+		conf = gitResolverConfig.ScmTokens["default"]
+	}
```

Comment:
This few lines are used 4 times in the code to extract a configuration that matches the config name provided in the parameters. In case the name is in the parameter, but it does not match the config, the `conf` object will be empty and we should error on that.

I would suggest moving this into a function that returns the error if applicable and use that function in the various places.

Reply:
ack , will do this change